classes/external/ai/expand_ai_text.php

```diff
@@ -119,23 +119,23 @@ public static function execute(
                 'assignment' => $assignment_id
             ]);
 
-            $generated_text = $ai_factory->openai()->api()->expand_ai_text(
+            $generated_text = $ai_factory->api()->expand_ai_text(
                 $step_1_additional_prompt === false ? '' : $step_1_additional_prompt,
                 $selected_text
             );
 
             $new_text = $ai_factory->formatter()->replace(
                 $text,
                 $selected_text,
-                $generated_text->get_text(),
+                $generated_text,
                 $select_start
             );
 
             $archive->commit_by_expand_ai_text(
                 $selected_text,
-                $generated_text->get_text(),
+                $generated_text,
                 $new_text,
-                $generated_text->get_response_json()
+                $generated_text
             );
```

Comment:
Couldn't one of the parameter in 'commit_by_expand_ai_text' be removed, since two of them have $generated_text?